### PR TITLE
Proprietary extension support

### LINKED ISSUE #161: Extensions v0.7.1 and above do not respond to commands
With Code extension version 0.7.0 and before, you were able to use "Lingua Franca: Build and Run" using Code's command palette. There was no effect of using these commands in the later versions on both WSL and OSX.

### PR DESCRIPTION
**Pull request by petervdonovan**
Fixes #161.

The problem was that this repository got out of sync with `lingua-franca`'s `master` branch. `lingua-franca`'s `master` branch incorporated changes from a "proprietary extension support" branch, but this companion branch had not been merged yet.

It feels necessary to explain how this happened since this problem can be avoided in the future. Part of the problem is that I spent a too much time working in unmerged feature branches and even declared that a feature had been implemented (while pointing to the unmerged feature branch) before the branch had been merged. This is a questionable practice that arose partly due to a desire to take a shortcut and partly due to uncertainty about what we actually would want in `main` in the long term. However, it is also true that we probably could have avoided publishing broken releases if I had been available to work in the past month.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,77 +10,99 @@ from the VSCode marketplace. See
...
+
 If you do not have a public key set up for authentication with GitHub, you can also use HTTPS:
+
 ```
 git clone https://github.com/lf-lang/vscode-lingua-franca.git \
 && cd vscode-lingua-franca \
 && npm install
 ```
 
+Note that this assumes that you have the WASM dependency somewhere on your system. Eventually, the WASM will probably be published; however, that hasn't been done yet, and it isn't currently possible to build it from source from this open-source repo because it is closed-source.
```
Comment (lhstrh) [inline CONTRIBUTING.md:30]: I guess this comment is now outdated?

Comment (petervdonovan) [reply CONTRIBUTING.md:30]: Yep, deleted it.

File: tsconfig.json [hunk 2]
```diff
@@ -1,15 +1,25 @@
 {
     "compilerOptions": {
-        "esModuleInterop": true,
+        // "esModuleInterop": true,
```
Comment (lhstrh) [inline tsconfig.json:3]: Why?

File: write-version-to-file.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+echo "'use strict';" > $1 && echo "// This is a generated file. Do not edit." >> $1 && echo "export const version = \"$(git ls-files | grep -v "$1" | git hash-object --stdin)\";" >> $1
```
Comment (lhstrh) [inline write-version-to-file.sh:1]: What is this used for?

Comment (petervdonovan) [reply write-version-to-file.sh:1]: It is used to write a hash of all tracked files to a file. This hash is used in a new "getVersion" command which is exposed by the VS Code extension. This command was to be used by the Xronos VS Code extension to check for compatibility. However, the command does not appear to be used at all currently. We could remove it.

File: tsconfig.json (hunk 2 above)
Comment (petervdonovan) [reply tsconfig.json:3]: I must admit that my understanding of this option is not good. While adding Wasm compatibility and switching over to Webpack, I found it necessary to change the output format of the bundler/transpiler to be Node16, which I think is more modern than CommonJS. My hand-wavy answer is that this probably made esModuleInterop unnecessary by converting everything to a common modules format.

File: write-version-to-file.sh (hunk 3 above)
Comment (lhstrh) [reply write-version-to-file.sh:1]: No need. Looks useful.

**Review by lhstrh — APPROVED** [review]
LGTM! Thank you so much for fixing this quickly, @petervdonovan.

**Pull request merged**